src/pds/cognito/export_userpool.py

```diff
@@ -0,0 +1,101 @@
+import sys
+
+import boto3
+import datetime
+import json
+
+page_size = 60
+region = 'us-west-2'
+
+
+# function to get user pool csv header
+def get_cognito_csv_header(cognito_client, user_pool_id):
+
+    result = None
+    csv_header_record = cognito_client.get_csv_header(UserPoolId = user_pool_id)
+    if 'CSVHeader' in csv_header_record:
+        result = csv_header_record['CSVHeader']
+
+    return result
+
+
+# function to get next page of users from the user pool
+def get_cognito_users(cognito_client, user_pool_id, next_pagination_token ='', limit = page_size):  
+
+    return cognito_client.list_users(
+        UserPoolId = user_pool_id,
+        Limit = limit,
+        PaginationToken = next_pagination_token
+    ) if next_pagination_token else cognito_client.list_users(
+        UserPoolId = user_pool_id,
+        Limit = limit
+    )
+  
+
+# extract attributes according to the given header list
+def extract_values(header_list, user_record):
+    # put the Username and Attributes in a form that's easier to search across
+    user_dict = {}
+    user_dict['cognito:username'] = user_record['Username']
+    for attr in user_record['Attributes']:
+        user_dict[attr['Name'].lower()] = attr['Value']
+
+    result = []
+    for header_name in header_list:
+        if header_name in user_dict:
+            result.append(user_dict[header_name])
+        else: 
+            result.append("")
+
+    return result
+
+
+# ensures that datetimes are handled as strings
+def datetimeconverter(o):
+    if isinstance(o, datetime.datetime):
+        return str(o)
+
+  
+#------------------------------
+# Process the cognito user pool
+#------------------------------
+
+if len(sys.argv) <= 1:
```

Comment:
Is it possible to add Cognito user pool also as an variable in the `variables.tf` file, so we can enter it using both as a variable in file (in cases of build automation, multiple variables in a file etc.) and also as a variable in command line (in case a user executes this interactively)?